### PR TITLE
Fixes #285, SEEK behavior.

### LINKED ISSUE #285: SEEK behavior
Currently SEEK will go to the next unit based on creation order. This sometimes results in jumping to units across the map when there are other units right near the cursor where the player's attention presumably is.

It would be far more intuitive for the cursor to jump to the nearest available unit instead. The unit seek-order should not change between back-to-back SEEK inputs so that the player can leave a local unit cluster if desired.

### PR DESCRIPTION
**Pull request by ThatIsAUsername**

### CODE REVIEW

File: src/Engine/MapController.java [hunk 1]
```diff
@@ -153,39 +154,39 @@ private void handleFreeTileSelect(InputHandler.InputAction input)
...
+          // First get all active units, sorted.
+          ArrayList<XYCoord> unitLocations = new ArrayList<XYCoord>();
+          unitLocations.addAll(Utils.findLocationsNearUnits(myGame.gameMap, myGame.activeCO.units, 0));
+          unitLocations.removeIf(xy -> myGame.gameMap.getLocation(xy).getResident().isTurnOver);
+          Utils.sortLocationsByDistance(myGame.getCursorCoord(), unitLocations);
+
+          // Find all usable properties, sorted.
+          ArrayList<XYCoord> usableProperties = Utils.findUsableProperties(myGame.activeCO, myGame.gameMap);
+          usableProperties.removeIf(xy -> myGame.gameMap.getLocation(xy).getResident() != null);
+          Utils.sortLocationsByDistance(myGame.getCursorCoord(), usableProperties);
```
Comment (ThislsAUsername) [inline src/Engine/MapController.java:172]: Om nom delicious lambda

**Comment by ThislsAUsername** [comment]
Having played around with this a bit, I think the distance-sorting should take precedence over unit vs industry.

File: src/Engine/MapController.java (hunk 1 above)
Comment (ThatIsAUsername) [reply src/Engine/MapController.java:172]: `Utils.sortLocationsByDistance` is a perfectly standard function, I assure you

**Comment by ThatIsAUsername** [comment]
I... can't say I'm a huge fan of that. I keep getting surprised by a build menu when I'm trying to move something.

File: src/Engine/MapController.java (hunk 1 above)
Comment (ThislsAUsername) [reply src/Engine/MapController.java:172]: I didn't wanna split the code block. I was referring to 171.

Comment (ThatIsAUsername) [reply src/Engine/MapController.java:172]: Yes.

**Comment by ThislsAUsername** [comment]
> I... can't say I'm a huge fan of that. I keep getting surprised by a build menu when I'm trying to move something.

IDK, I tried it and it worked nicely IMO. Maybe make it an option?

**Comment by ThatIsAUsername** [comment]
> Maybe make it an option?

Ehhh, I hate to turn everything into an option. I mean, if you really want to we could, but I wouldn't otherwise.

**Comment by ThislsAUsername** [comment]
> Ehhh, I hate to turn everything into an option. I mean, if you really want to we could, but I wouldn't otherwise.

Agreed on that. You strongly prefer production last, I strongly prefer production mixed in... seems like it'd make a decent option.

**Pull request merged**